### ISSUE TITLE
limitRecent is confusing

### ISSUE THREAD

**Issue by mpickering** [nomeata/gipeda#29]
I assumed that the limitRecent option in settings.yaml referred to how many commits were shown on the graph. However, it doesn't affect that at all and it just limits the number of commits shown on the homepage. 

To modify the total number of commits shown on the graph you have to edit the constant `cGRAPH_HISTORY` in `Shake.hs`. It would be good to be able to control this from the config file as well.

**Comment by mpickering** [comment]
and even when I set that constant to a very large number, I can't get all the commits to show up on the graph.

**Comment by nomeata** [comment]
> To modify the total number of commits shown on the graph you have to edit the constant cGRAPH_HISTORY in Shake.hs. It would be good to be able to control this from the config file as well.

Yes, this is most likely an oversight by me. Or maybe I was unsure about how to get a configuration value at that point.

> and even when I set that constant to a very large number, I can't get all the commits to show up on the graph. 

I won’t show all, but only the chain of parents from the latest commit. Would that explain your observation?

**Comment by mpickering** [comment]
I was using the example site generated by the scripts in the `example` directory. 

```
commit 1baeb582e24b56f15abcedaaf1028d1655b9f3b0
Author: Joachim Breitner <mail@joachim-breitner.de>
Date:   Tue Jan 20 00:07:57 2015 +0100

    Do not cache ajax requests

    We cache them while the page is open, otherwise we really want the
    latest (especially for latest-summaries.json).
```

This was the first commit being shown.

**Comment by mpickering** [comment]
I actually think this fact is confused in certain places in the code. If you look at calls to `commitsFrom` in the javascript logic then `limitRecent` is used in places which are nothing to do with the homepage.
